File: EE3015-EE3025/Assignment-2/Codes/MatLab Codes/iir/add.m
function z = add(x,y)
  m = length(x);
  n = length(y);

  if m == n
      z = x + y;
  elseif m > n
      z = x + [zeros(1,m-n) y];
  else
      z = [zeros(1,n-m) x] + y;
end

File: EE3015-EE3025/Assignment-2/Report/codes/iir/add.py
import numpy as np

def add(x,y):
	x = np.array(x)
	y = np.array(y)
	
	m = x.shape[0]
	n = y.shape[0]
	
	if m == n:
		z = x + y
	elif m > n:
		z = x + np.append(np.zeros((m-n,)), y)
	else:
		z = y + np.append(np.zeros((n-m,)), y)
	
	return z
